server/controllers/messages.js

```diff
@@ -25,7 +30,14 @@ const hasRead = (message,id) => {
 
 export default {
 
+  /**
+   * api route to send a message to a group
+   * @param {object} req 
+   * @param {object} res 
+   * @return {void}
+   */
   sendMessage(req, res) {
+    // authentication
     if (req.header('x-auth')) {
       const token = req.header('x-auth');
       jwt.verify(token, secret, (err, decoded) => {
```

Comment:
It is better you create a middleware that protects your app from non logged in user sending messages